pyEpiabm/pyEpiabm/routine/file_population_config.py

```diff
@@ -95,14 +101,14 @@ def make_pop(input_file: str, random_seed: int = None, time: float = 0):
                 cell.set_location(location)
 
             # Raise error if microcell exists, then create new one
-            for microcell in cell.microcells:
-                if microcell.id == line["microcell"]:
-                    raise ValueError(f"Duplicate microcells {microcell.id}"
-                                     + f" in cell {cell.id}")
+            microcell_ids = [microcell.id for microcell in cell.microcells]
+            if microcell_id_csv in microcell_ids:
+                raise ValueError(f"Duplicate microcells {microcell_id_csv}"
```

Comment:
This doesn't read quite right once you substitute in the values in {} - maybe 'Duplicate microcells: {...} already exists in cell {...}'?